src/uwtools/config.py

```diff
@@ -104,23 +106,47 @@ def dump_file(self, output_path):
         with open(output_path, 'w', encoding="utf-8") as file_name:
             yaml.dump(self.data, file_name, sort_keys=False)
 
+    class Loader(yaml.SafeLoader): #pylint: disable=too-many-ancestors
+        '''Yaml loader to add INCLUDE tag for yaml file concatenation'''
+
+        def __init__(self, stream: IO) -> None:
+            try:
+                self._root = os.path.split(stream.name)[0]
+            except AttributeError:
+                self._root = os.path.curdir
+
+            super().__init__(stream)
```

Comment:
It is unclear to me the usefulness of this local class here. What benefit do we gain from making this class? It seems that you have added the constructor from a method of the YAMLConfig class. Why not remove the Loader class and use SafeLoader directly? For example:

```
yaml.add_constructor("!INCLUDE", add_include, Loader=yaml.SafeLoader)
```